components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Cdm/src/Services/CdmMessageProcessor.cs

```diff
@@ -217,16 +251,36 @@ public class CdmMessageProcessor : ICdmClient {
             sw.Stop();
         }
 
-        private string GetNormalizedEntityName(string publisherId,
-            string dataSetWriterId, string metadataVersion) {
-            var key = string.Join("", publisherId.Split(Path.GetInvalidFileNameChars())) + '_' +
-                      string.Join("", dataSetWriterId.Split(Path.GetInvalidFileNameChars()));
-            if (!string.IsNullOrEmpty(metadataVersion)) {
-                key += '_' + string.Join("", metadataVersion.Split(Path.GetInvalidFileNameChars()));
+        /// <inheritdoc/>
+        private string GetNormalizedEntityName(MonitoredItemSampleModel sample) {
+            if (string.IsNullOrEmpty(sample.PublisherId) ||
+                string.IsNullOrEmpty(sample.DataSetWriterId) ||
+                string.IsNullOrEmpty(sample.NodeId)) {
+                return null;
             }
-            return key.Replace('#', '_').Replace('.', '_').Replace('/','_').Replace(':', '_').Replace('"', '_').Replace('\'', '_');
+            var key = $"{sample.PublisherId}_{sample.DataSetWriterId}" +
+                $"_{sample.NodeId}";
+            key = string.Join("", key.Split(Path.GetInvalidFileNameChars()));
+            return key.Replace('#', '_').Replace('.', '_')
+                .Replace('/', '_').Replace(':', '_')
+                .Replace('"', '_').Replace('\'', '_');
```

Comment:
code duplication in the function below, this calls for a small helper func

Reply:
thanks!